R/srcrefs.R

```diff
@@ -139,17 +139,47 @@ srcrefs.R6ClassGenerator <- function(x, ..., srcref_names = NULL,
 
 ### S4
 
+#' @exportS3Method
+#' @importFrom utils getSrcref
+#' @rdname srcrefs
+srcrefs.standardGeneric <- function(x, ..., srcref_names = NULL) {
+  if (is.null(sr <- getSrcref(x))) return(sr)
+
+  attr(sr, "namespace") <- obj_namespace_name(x)
+
+  objs <- list(sr)
+  names(objs) <- srcref_names
+  objs
+}
+
+#' @exportS3Method
+#' @importFrom utils getSrcref
+#' @rdname srcrefs
+srcrefs.nonstandardGenericFunction <- srcrefs.standardGeneric
```

Comment:
specialized srcref discovery for S4 generic functions and binding of the namespace as a srcref attribute